WeakAurasOptions/AceGUI-Widgets/AceGUIWidget-WeakAurasSnippetButton.lua

```diff
@@ -0,0 +1,228 @@
+  self:ClearFocus()
+  AceGUI:ClearFocus()
+  self:Disable()
+  self:Hide()
+  self:GetParent().obj:Fire("OnEnterPressed", ...)
+end
+--[[-----------------------------------------------------------------------------
+Methods
+-------------------------------------------------------------------------------]]
+local methods = {
+  ["OnAcquire"] = function(self)
+    -- restore default values
+    self:SetDisabled(false)
+    self:SetTitle()
+    self:SetEditable(false)
+  end,
+  -- ["OnRelease"] = nil,
+
+  ["SetTitle"] = function(self, text)
+    self.frame.titleText = text
+    self.title:SetText(text)
+  end,
+  ["SetDescription"] = function(self, text)
+    self.frame.descriptionText = text
+  end,
+  ["SetDisabled"] = function(self, disabled)
+    self.disabled = disabled
+    if disabled then
+      self.frame:Disable()
+    else
+      self.frame:Enable()
+    end
+  end,
+  ["LockHighlight"] = function(self)
+    self.frame:LockHighlight()
+  end,
+  ["UnlockHighlight"] = function(self)
+    self.frame:UnlockHighlight()
+  end,
+  ["SetEditable"] = function(self, bool)
+    if bool then
+      self.frame.editable = true
+      self.deleteButton:Show()
+      self.title:SetPoint("LEFT", self.deleteButton, "RIGHT")
+    else
+      self.frame.editable = false
+      self.deleteButton:Hide()
+      self.title:SetPoint("LEFT", self.deleteButton, "LEFT", 4, 0)
+    end
+  end
+}
+
+--[[-----------------------------------------------------------------------------
+Constructor
+-------------------------------------------------------------------------------]]
+local function Constructor()
+  local name = "WeakAurasSnippetButton" .. AceGUI:GetNextWidgetNum(Type)
+  local button = CreateFrame("Button", name, UIParent, "OptionsListButtonTemplate")
+  button:Hide()
+
+  button:EnableMouse(true)
+  button:SetScript("OnClick", Button_OnClick)
+  button:SetScript("OnEnter", Control_OnEnter)
+  button:SetScript("OnLeave", Control_OnLeave)
+
+  button:SetHeight(24)
+  button:SetWidth(170)
+
+  local deleteButton = CreateFrame("BUTTON", nil, button)
+  deleteButton:SetPoint("LEFT", button, "LEFT")
+  deleteButton:SetSize(20, 20)
+  local deleteTex = deleteButton:CreateTexture()
+  deleteTex:SetAllPoints()
+  deleteTex:SetTexture([[Interface\Buttons\CancelButton-Up]])
+  deleteTex:SetTexCoord(0.1, 0.9, 0.1, 0.9)
+  deleteButton:SetNormalTexture(deleteTex)
+  deleteButton:Hide()
+  button.deleteButton = deleteButton
+
+  local title = button:CreateFontString(nil, "OVERLAY", "GameFontHighlightLarge")
+  title:SetHeight(14)
+  title:SetJustifyH("LEFT")
+  title:SetPoint("LEFT", deleteButton, "RIGHT")
+  title:SetPoint("RIGHT", button, "RIGHT")
+  title:SetTextColor(1, 1, 1, 1)
+  button.title = title
+
+  local ntex = button:CreateTexture()
+  ntex:SetTexture("Interface\\BUTTONS\\UI-Listbox-Highlight2.blp")
+  ntex:SetVertexColor(0.8, 0.8, 0.8, 0.25)
+  ntex:SetPoint("TOPLEFT", 1, -1)
+  ntex:SetPoint("BOTTOMRIGHT", -1, 1)
```

Comment:
Something about this must be incorrect, there's a margin on the left but not on the right.

Reply:
This should be 1 pix in, in each direction. Though left and right don't actually matter so I've set them to 0.

Reply:
Yes, it should. But take a look at how it actually looks like. For me the buttons are not centered.